### PR TITLE
fix: fix several situations where we were incorrectly inferring the storage version

### PR DESCRIPTION
**Pull request by westonpace**
This could lead to corrupt datasets with a mix of v1 and v2 files.

### CODE REVIEW

File: rust/lance-table/src/format/manifest.rs [hunk 1]
```diff
@@ -436,10 +436,16 @@ impl TryFrom<pb::Manifest> for Manifest {
...
             None => {
-                if has_deprecated_v2_feature_flag(p.writer_feature_flags) {
-                    DataStorageFormat::new(LanceFileVersion::V2_0)
+                if fragments.is_empty() {
+                    // No fragments to inspect, best we can do is look at writer flags
+                    if has_deprecated_v2_feature_flag(p.writer_feature_flags) {
+                        DataStorageFormat::new(LanceFileVersion::Legacy)
+                    } else {
+                        DataStorageFormat::new(LanceFileVersion::Stable)
+                    }
```
Comment (wjones127) [inline rust/lance-table/src/format/manifest.rs:445]: Isn't this backwards?

Comment (westonpace) [reply rust/lance-table/src/format/manifest.rs:445]: Good point.  Let me make some migration tests for these cases

Comment (westonpace) [reply rust/lance-table/src/format/manifest.rs]: Ok.  I added some regression test cases (confirmed they detected this) and switched the logic (now tests pass)

File: python/python/tests/historical_datasets/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Historical Datasets
+
+This folder cotnains datasets from older versions of Lance that we use for backwards
```
Comment (wjones127) [inline python/python/tests/historical_datasets/README.md:3]:
```suggestion
This folder contains datasets from older versions of Lance that we use for backwards
```

File: python/python/tests/historical_datasets/README.md
Comment (wjones127) [inline python/python/tests/historical_datasets/README.md:1]: Can we move these to the `test_data` folder, like the other datasets? https://github.com/lancedb/lance/tree/main/test_data

**Review by wjones127 — CHANGES_REQUESTED** [review]

File: python/python/tests/historical_datasets/README.md
Comment (westonpace) [reply python/python/tests/historical_datasets/README.md:1]: Done

**Review by wjones127 — APPROVED** [review]
Thanks!

File: python/python/tests/historical_datasets/README.md (hunk 2 above)
Comment (broccoliSpicy) [reply python/python/tests/historical_datasets/README.md]: maybe we could introduce spelling check for our repo? such as https://github.com/crate-ci/typos

**Pull request merged**